File: src/backend/reservation/src/main/java/org/taonaw/reservation/domain/model/equipment/Equipment.java
package org.taonaw.reservation.domain.model.equipment;

import lombok.NonNull;

import java.util.Objects;

public class Equipment {
    private final EquipmentId equipmentId;
    private final EquipmentStocks equipmentStocks;

    public Equipment(@NonNull EquipmentId equipmentId,
                     @NonNull EquipmentStocks equipmentStocks) {
        this.equipmentId = equipmentId;
        this.equipmentStocks = equipmentStocks;
    }

    public EquipmentId getEquipmentId() {
        return equipmentId;
    }

    public EquipmentStocks getEquipmentStocks() {
        return equipmentStocks;
    }

    public boolean isOutOfStocks(int reservedEquipmentQuantity) {
        return equipmentStocks.isOutOfStocks(reservedEquipmentQuantity);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Equipment equipment = (Equipment) o;
        return equipmentId.equals(equipment.equipmentId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(equipmentId);
    }
}

File: src/backend/reservation/src/main/kotlin/org/taonaw/reservation/domain/model/equipment/Equipment.kt
package org.taonaw.reservation.domain.model.equipment

import org.taonaw.reservation.domain.model.shared.Fee

class Equipment(
        val equipmentId: EquipmentId,
        val equipmentStocks: EquipmentStocks,
        val rentalFee: Fee) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Equipment

        if (equipmentId != other.equipmentId) return false

        return true
    }

    override fun hashCode(): Int {
        return equipmentId.hashCode()
    }
}